pageserver/src/tenant/timeline/eviction_task.rs

```diff
@@ -81,6 +81,12 @@ impl Timeline {
     #[instrument(skip_all, fields(tenant_id = %self.tenant_shard_id.tenant_id, shard_id = %self.tenant_shard_id.shard_slug(), timeline_id = %self.timeline_id))]
     async fn eviction_task(self: Arc<Self>, cancel: CancellationToken) {
         use crate::tenant::tasks::random_init_delay;
+
+        // acquire the gate guard only once within a useful span
+        let Ok(guard) = self.gate.enter() else {
+            return;
+        };
+
```

Comment:
The `random_init_delay` below isn't sensitive to guard cancellation. Not sure if it matters. Do we have a helper ot merge guard and CancellationToken cancellation? I don't think so, otherwise page_service.rs would look better.

Same for the `timeout_at` below.

Reply:
> Gate guards don't have cancellation in general, rightly so.

Riight. Allright then,.